packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panels_builder.dart

```diff
@@ -15,6 +15,7 @@ class KnowledgePanelsBuilder {
     KnowledgePanels knowledgePanels, {
     final Product? product,
     final BuildContext? context,
+    final List<String>? panelIds,
```

Comment:
this makes the contract of this file a little complicated, previously this was building all knoweldge panels in `root` now it can selectively build just one if needed. 

I think we can live without it, why don't you just call `KnowledgePanelElementCard` from the caller ?

Reply:
or `KnowledgePanelCard`

Reply:
Oh it's optional. And we cannot live without it.
We need it because now I use only one single "fair" knowledge panel data file (with both health and environment card). The previous version used 2 edited files from assets, one with health and one with environment data. We cannot do that now, because we're downloading the (localized) data from the back-end.
Does that make sense?

If it's just the syntax that you don't like, I'm open to suggestions, but we need an optional way to filter cards.

Reply:
Got it. let me quickly patch this PR and see if there's another way.

Reply:
There would be another way if in `KnowledgePanelsBuilder.build` we used a specific method (let's call it `toto`, I'm French) to build each card - and I would call directly `toto` for each onboarding screen. Why not.
We can do that later.

Reply:
Another idea would be to create a helper function to get KnowledgePanelElement

Something like:

```

  KnowledgePanelElement? getKnowledgePanelElementFromRoot(
      KnowledgePanels knowledgePanels,
      String panelId) {
    for (final KnowledgePanelElement panelElement
    in knowledgePanels.panelIdToPanelMap['root']!.elements!) {
      if (panelElement.elementType != KnowledgePanelElementType.PANEL) {
        continue;
      }
      final String panelElementId = panelElement.panelElement!.panelId;
      if (panelElementId != panelId) {
        continue;
      }
      return panelElement;
    }
    return null;
  }
}
```

And then build `KnowledgePanelElementCard` widget

Reply:
It feels a bit cleaner, from the perspective of a reader `if (panelIds != null && !panelIds.contains(panelId)) ` part feels confusing

Reply:
@jasmeet0817 I've just pushed cleaner code, with explicit `buildAll` and `buildSingle` methods regarding panels.